src/compute-client/src/controller/instance.rs

```diff
@@ -1140,6 +1140,14 @@ where
         response: SubscribeResponse<T>,
         replica_id: ReplicaId,
     ) -> Option<ComputeControllerResponse<T>> {
+        // We should not receive updates for subscribes we don't track. It is possible that we
```

Comment:
This comment and the following comment seem to contradict each other? I think the problem emerges from the distinction of tracking collections vs. tracking subscribes. We probably should say here something like `We should not receive updates for _collections_ we don't track`. This is the exact same formulation in `handle_frontier_uppers`, which would be fair given that the problem referred to there is the same as the one referred to here?

Reply:
Good catch! I'll amend this in a follow-up.

Reply:
I added that change to https://github.com/MaterializeInc/materialize/pull/17540.